### PR TITLE
Wi-Fi shell improvs

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

**Review by rado17 — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

File: samples/wifi/shell/README.rst [hunk 1]
```diff
@@ -115,11 +115,15 @@ Supported CLI commands
        | [-h, --help] : Print out the help for the scan command.
    * - connect
      - | Connect to a Wi-Fi AP with the following parameters:
-       | <SSID>
-       | <Channel number> (optional: 0 means all)
-       | <PSK> (optional: valid only for secured SSIDs)
-       | <Security type> (optional: 0-None, 1-PSK, 2-PSK-256, 3-SAE)
-       | <MFP> (optional: 0-Disable, 1-Optional, 2-Required)
+       | "<SSID>"
+       | [channel number/band: > 0:Channel, 0:any channel,
```
Comment (richabp) [inline samples/wifi/shell/README.rst:119]: Is it required to mention it twice 0:Channel, 0:Any channel?
```suggestion
       | [Channel number/band: > 0:Channel, 0:Any channel,
```

File: samples/wifi/shell/README.rst [hunk 2]
```diff
@@ -115,11 +115,15 @@ Supported CLI commands
...
+       | < 0:band (-2:2.4GHz, -5:5GHz, -6:6GHz]
+       | [PSK: valid only for secure SSIDs]
+       | [Security type: valid only for secure SSIDs]
+       | 0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:EAP, 6:WEP, 7:
+       | WPA-PSK
+       | [MFP (optional: needs security type to be specified)]
+       | : 0:Disable, 1:Optional, 2:Required.
```
Comment (richabp) [inline samples/wifi/shell/README.rst:126]:
```suggestion
       | 0:Disable, 1:Optional, 2:Required.
```

File: samples/wifi/shell/README.rst [hunk 3]
```diff
@@ -133,8 +137,14 @@ Supported CLI commands
        | <SSID length>
        | <channel> [optional]
        | <psk> [optional]
+       |
        | disable - Disable Access Point mode
        | (Note that the Access Point mode is presently not supported.)
+       |
+       | stations  : List stations connected to the AP
```
Comment (richabp) [inline samples/wifi/shell/README.rst:144]:
```suggestion
       | stations  - List stations connected to the AP
```

File: samples/wifi/shell/README.rst [hunk 4]
```diff
@@ -147,19 +157,22 @@ Supported CLI commands
...
      - | Manage Target Wake Time (TWT) flows with below subcommands:
        |
-       | setup - Start a TWT flow:
-       | <negotiation_type: 0 - Individual, 1 - Broadcast, 2 - Wake TBTT>
-       | <setup_cmd: 0 - Request, 1 - Suggest, 2 - Demand>
-       | <dialog_token: 1-255> <flow_id: 0-7> <responder: 0/1> <trigger: 0/1>
-       | <implicit: 0/1> <announce: 0/1> <twt_wake_interval: 1-262144 µs>
-       | <twt_interval: 1µs-2^31µs>
+       | quick_setup   : Start a TWT flow with defaults:
+       |  <twt_wake_interval: 1-262144us> <twt_interval: 1us-2^31us>.
```
Comment (richabp) [inline samples/wifi/shell/README.rst:161]:
```suggestion
       | quick_setup: Start a TWT flow with defaults -
       | <twt_wake_interval: 1 µs - 262144 µs> <twt_interval: 1 µs - 2^31 µs>
```

File: samples/wifi/shell/README.rst [hunk 5]
```diff
@@ -147,19 +157,22 @@ Supported CLI commands
...
+       |
+       | setup         : Start a TWT flow:
+       |  <negotiation_type, 0: Individual, 1: Broadcast, 2: Wake TBTT>
+       |  <setup_cmd: 0: Request, 1: Suggest, 2: Demand>
+       |  <dialog_token: 1-255> <flow_id: 0-7> <responder: 0/1> <trigger:
+       |  0/1> <implicit:0/1> <announce: 0/1> <twt_wake_interval:
+       |  1-262144us> <twt_interval: 1us-2^31us>.
```
Comment (richabp) [inline samples/wifi/shell/README.rst:168]:
```suggestion
       | setup: Start a TWT flow -
       | <negotiation_type, 0: Individual, 1: Broadcast, 2: Wake TBTT>
       | <setup_cmd: 0: Request, 1: Suggest, 2: Demand>
       | <dialog_token: 1-255> <flow_id: 0-7> <responder: 0/1> <trigger: 0/1>
       | <implicit: 0/1> <announce: 0/1> <twt_wake_interval: 1 µs - 262144 µs>
       | <twt_interval: 1 µs - 2^31 µs>
```

File: samples/wifi/shell/README.rst [hunk 6]
```diff
@@ -147,19 +157,22 @@ Supported CLI commands
...
        |
-       | teardown - Teardown a TWT flow:
-       | <negotiation_type: 0 - Individual, 1 - Broadcast, 2 - Wake TBTT>
-       | <setup_cmd: 0 - Request, 1 - Suggest, 2 - Demand>
-       | <dialog_token: 1-255> <flow_id: 0-7>
+       | teardown      : Teardown a TWT flow:
+       |  <negotiation_type, 0: Individual, 1: Broadcast, 2: Wake TBTT>
+       |  <setup_cmd: 0: Request, 1: Suggest, 2: Demand>
+       |  <dialog_token: 1-255> <flow_id: 0-7>.
```
Comment (richabp) [inline samples/wifi/shell/README.rst:173]:
```suggestion
       | teardown: Teardown a TWT flow -
       | <negotiation_type, 0: Individual, 1: Broadcast, 2: Wake TBTT>
       | <setup_cmd: 0: Request, 1: Suggest, 2: Demand>
       | <dialog_token: 1-255> <flow_id: 0-7>
```

File: samples/wifi/shell/README.rst [hunk 7]
```diff
@@ -147,19 +157,22 @@ Supported CLI commands
...
        |
-       | teardown_all - Teardown all TWT flows
+       | teardown_all  : Teardown all TWT flows.
```
Comment (richabp) [inline samples/wifi/shell/README.rst:175]:
```suggestion
       | teardown_all  : Teardown all TWT flows
```

File: samples/wifi/shell/README.rst [hunk 8]
```diff
@@ -192,6 +205,29 @@ Supported CLI commands
        | wifi mode -g -i1
        | Set operation example for interface index 1 - set station+promiscuous
        | wifi mode -i1 -sp
+   * - packet_filter
+     - | This command is used to set packet filter setting when
+       | monitor, TX-Injection and promiscuous mode is enabled
+       | The different packet filter modes are control,
+       | management, data and enable all filters
```
Comment (richabp) [inline samples/wifi/shell/README.rst:212]: TX injection mode is edited as per https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/wifi/raw_tx_operation.html#tx-injection-mode for consistency.
```suggestion
   * - packet_filter
     - | This command is used to set packet filter settings when
       | monitor, TX injection, and promiscuous mode are enabled.
       | The different packet filter modes are control,
       | management, data, and enable all filters.
```

**Review by richabp — CHANGES_REQUESTED** [review]
Some minor style issues, hope it can be fixed.

File: samples/wifi/shell/README.rst (hunk 1 above)
Comment (krish2718) [reply samples/wifi/shell/README.rst:119]: It's `> 0`  and `0`, both are different.

**Comment by krish2718** [comment]
@richabp  the text here is not just docs, it's from the code at https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/l2/wifi/wifi_shell.c#L1866. This PR is just trying to sync both.

**Review by VivekUppunda — APPROVED** [review]

**Comment by krish2718** [comment]
> @richabp  the text here is not just docs, it's from the code at https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/l2/wifi/wifi_shell.c#L1866. This PR is just trying to sync both.

I can raise an upstream PR to fix the help or find a way to remove this duplication by linking to upstream docs in a new PR.

**Pull request merged**